wrapper/src/main/java/software/amazon/jdbc/ConnectionPlugin.java

```diff
@@ -32,28 +34,28 @@ public interface ConnectionPlugin {
   Set<String> getSubscribedMethods();
 
   <T, E extends Exception> T execute(
-      final Class<T> resultClass,
-      final Class<E> exceptionClass,
-      final Object methodInvokeOn,
-      final String methodName,
-      final JdbcCallable<T, E> jdbcMethodFunc,
-      final Object[] jdbcMethodArgs)
+      final @Nullable Class<T> resultClass,
```

Comment:
I believe that @NonNull makes more sense here.

Reply:
Agreed, but I'm fairly certain we call it somewhere with a null